lib/puppet/functions/cd4pe_deployments/get_impact_analysis_reports.rb

```diff
@@ -0,0 +1,55 @@
+require 'puppet_x/puppetlabs/cd4pe_client'
+require 'puppet_x/puppetlabs/cd4pe_function_result'
+
+# @summary Get all the Impact Analysis reports for a commit in the current pipeline
+Puppet::Functions.create_function(:'cd4pe_deployments::get_impact_analysis_reports') do
+  # @param commit_sha
+  #   The commit SHA used in the Impact Analysis reports
+  # @example Get information about a specific environment in a specific IA
+  #   $impact_analysis_reports = get_impact_analysis_reports('47b552dc15448b4e306fcf8df320e5124b2cbd63', 'control-repo')
+  # @return [Array] contains all Impact Analysis reports found in the pipeline
+  #   See [README.md]() for information on the CD4PEFunctionResult hash format
+  #   * result [Hash]:
+  #     * `rows [Array] array of hashes, one for each report`
+  #   * error [Hash] contains error information if any
+  #
+  dispatch :get_impact_analysis_reports do
+    required_param 'String', :commit_sha
+    required_param 'String', :repo_name
+  end
+
+  def get_impact_analysis_reports(commit_sha, repo_name)
```

Comment:
I love how you've leveraged existing codez to get the reports. 🎉 

Not that we necessarily want to tackle this now, but I'm curious if you or others have thoughts about recreating this as a new endpoint in the Java codebase to allow this client to make one single network call, and to reduce what this client needs to know about CD4PE to get the information it needs. If we did want to do that, we could replace the guts of this ruby code with a single network call, later.

Reply:
That would be nice.

Reply:
I agree. This is a really good way to quickly get new functionality in to custom deployment policies, but in the long term we shouldn't be adding too much logic in the client.